src/interactive-auth.ts

```diff
@@ -510,12 +578,13 @@ InteractiveAuth.prototype = {
         // information such that the app can give a better one if so desired.
         const err = new Error("No appropriate authentication flow found");
         err.name = 'NoAuthFlowFoundError';
-        err.required_stages = [];
-        if (haveEmail) err.required_stages.push(EMAIL_STAGE_TYPE);
-        if (haveMsisdn) err.required_stages.push(MSISDN_STAGE_TYPE);
-        err.available_flows = flows;
+        const requiredStages: string[] = [];
+        if (haveEmail) requiredStages.push(EMAIL_STAGE_TYPE);
+        if (haveMsisdn) requiredStages.push(MSISDN_STAGE_TYPE);
+        (err as any).required_stages = requiredStages;
+        (err as any).available_flows = flows;
```

Comment:
we should type the error, or at least leave comments about the cast. `// TODO: Types` is fine imo